### PR TITLE
Fast sync state error

### LINKED ISSUE #3227: Optional empty not handled: fastSyncState.getPivotBlockHeader().get();
Sonar is reporting a possible bug that looks worth fixing https://sonarcloud.io/project/issues?id=hyperledger_besu&open=AX3clLLkTSMTbb40goXd&resolved=false&sinceLeakPeriod=true&types=BUG

fastSyncState.getPivotBlockHeader().get();

We probably just want to log a message and exit if pivot block not present

### PR DESCRIPTION
**Pull request by TheSwarnim**
## PR description
Previously, fastSyncState.getPivotBlockHeader().get() is accessed before checking pivot block not present. Added fixes, needed to add the exit code but not able to understand.

## Fixed Issue(s)

Fixes #3227

## Changelog
Added the error message log if pivot block not present.

### CODE REVIEW

**Comment by TheSwarnim** [comment]
@macfarla, I've added some code. Please review it. Need some help!

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java:112]: We would need to exit this method if no pivot block. either return Optional.empty or throw an exception

**Comment by macfarla** [comment]
Hi @TheSwarnim left a comment but also you need to add signoff to your commits to keep DCO happy https://github.com/hyperledger/besu/pull/3261/checks?check_run_id=4764675277

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java
Comment (TheSwarnim) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java:112]: Ok, thanks for suggestion

**Comment by TheSwarnim** [comment]
Hi @macfarla, please review it again. I've added all the relevant changes.

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java:114]: this completeExceptionally looks right to me

**Review by macfarla — COMMENTED** [review]
your changes look good now but you have some conflicts. might need to do a rebase!

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java
Comment (TheSwarnim) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/FastWorldStateDownloader.java:114]: Ok, I'll check it.
Thanks

**Comment by TheSwarnim** [comment]
@ macfarla, please re-review it. I've resolved all the conflicts.

**Comment by macfarla** [comment]
Hi! Almost there! Now there's just some formatting to fix up:
 Run './gradlew :spotlessApply' to fix these violations.

**Comment by TheSwarnim** [comment]
@macfarla , please see it again. Hope I've completed all the steps successfully.
Thanks

**Comment by macfarla** [comment]
Hi @TheSwarnim getting close! Your last commit is missing signoff - https://github.com/hyperledger/besu/pull/3261/checks?check_run_id=4839549654 - the instructions are there to fix it up.

**Comment by TheSwarnim** [comment]
Hello @macfarla, I've signoff all commits. Please review it again.
Thanks

**Review by macfarla — APPROVED** [review]

**Comment by TheSwarnim** [comment]
Hi @macfarla, I think the error remains the same. I've checked the bug, it is showing almost the same error. I think I should recommit with some changes. Please review it. 
Also, I would also like to interact with you or the project maintainer so that I can get involved in the project more effectively and contribute to the project.
Thanks

**Comment by macfarla** [comment]
@TheSwarnim have a look at https://github.com/hyperledger/besu/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

and you can chat to maintainers here https://chat.hyperledger.org/channel/besu

**Pull request merged**
